### DISCUSSION TITLE
Spring Profile-Specific Configurations in Peppol-reporting

### DISCUSSION THREAD

**Discussion by MatteoBenvenuti** [phax/peppol-reporting#9] [Q&A]
Hello, I'm currently utilizing this library within a Spring application. I've configured the database connection settings in the 'application.properties' file, and they function correctly. However, I'm encountering an issue when implementing Spring profile management. Despite having an active profile, the library continues to read the configuration solely from the 'application.properties' file, neglecting the profile-specific settings.

I'm curious whether I might be misunderstanding something or if this behavior aligns with expectations. Is there a recommended method to ensure the library reads configurations specific to the active profile? Currently, I've devised a workaround using the MultiConfigurationValueProvider class and the addConfigurationSource() method, but I suspect this might not be the intended approach.

**Comment by phax** [comment]
Well, because I am not using Spring, I am also not using the configuration system of Spring :)
My configuration system is described in https://github.com/phax/ph-commons/wiki/ph-config and currently doesn't support such kind of profiling yes.
Instead my assumption was, that you create your `IConfig` object for tests etc. However, I see that question popping up quite some times, so I might consider implementing this, but I need to understand the consequences first and created https://github.com/phax/ph-commons/issues/40 for that.

My approach for local test properties is to use the file `private-application.properties` and add the specific local configuration properties there. Because properties in this file have higher precedence, they will be taken instead of the same property in `application.properties`. Additionally, to avoid confusion on deployment, I recommend that you ignore any `private-application.properties` in your git/svn repository.
